### ISSUE TITLE
rPLD cross validation procedure.

### ISSUE THREAD

**Issue by odemangeon** [rodluger/everest#16]
If I didn't made any mistake in my interpretation of the code, there seems to be a puzzling difference between what the code does and what the Luger+2017 paper says that the code is doing regarding the cross validation of rPLD. In the paper, it's mentioned that "we split each light curve segment into three roughly equal sections. For each pair of sections, we train the model on them and compute the model prediction in the third section (the validation set)".
From the code below, it looks like the training set is actually the whole chunk and the validation set is a pair of sections within the chunk.

https://github.com/rodluger/everest/blob/6779591f9f8b3556847e2fbf761bdfac7520eaea/everest/detrender.py#L545-L558

Did I misunderstood the code ?
If no, this way of selecting validation and training sets is strange because it implies that the validation set is actually 2/3 of training set.  The validation set should in principle be independent from the training set, right ? Is there a reason to do this anyway ?

**Comment by rodluger** [comment]
I don't think there's a bug, just very misleading names for things in the code. You're absolutely right that the training and validation sets should be independent: that's the whole point of cross-validation. And you're right that when I'm computing the model for the "training" set, I'm actually using the entire chunk. But I never actually do anything with what I'm calling the "training" set -- it's just used for plotting purposes; the scatter in the "training" set versus regularization strength is the blue line in the [diagnostic plots](https://archive.stsci.edu/hlsps/everest/v2/c01/201300000/67065/hlsp_everest_k2_llc_201367065-c01_kepler_v2.0_dvs.pdf) I publish on MAST.

The thing I actually use to optimize lambda is what I'm calling the "validation" set. When I pre-compute the matrices for this set, I am doing what I claim in the paper: masking one-third of the chunk and training on the other two-thirds. Then, on this line, I compute the scatter in the 1/3 portion of the light curve that was masked based on the model that was trained on the other 2/3. That scatter as a function of lambda is plotted as the red lines in the [diagnostic plots](https://archive.stsci.edu/hlsps/everest/v2/c01/201300000/67065/hlsp_everest_k2_llc_201367065-c01_kepler_v2.0_dvs.pdf); notice that the minimum always occurs at much smaller lambda than in the blue curve, since it's actually gauging the predictive power of the model.

So, long story short: *ignore* what I'm calling the "training" set, and focus on what I'm calling the "validation" set. Within the computations of the latter, I mask 1/3 of the chunk, compute the model based on the other 2/3, and compute the scatter of the prediction in the masked 1/3.

Again, I don't think there's anything wrong with the code -- I just shouldn't have called the red and blue curves what I did.
